src/pipeline.ts

```diff
@@ -186,17 +191,27 @@ export async function checkNewCandidateMatch(
 }
 
 async function findSearchableTorrents() {
-	const { torrents } = getRuntimeConfig();
+	const { torrents, dataDirs } = getRuntimeConfig();
 	let parsedTorrents: Searchee[];
 	if (Array.isArray(torrents)) {
 		const searcheeResults = await Promise.all(
-			torrents.map(createSearcheeFromTorrentFile)
+			torrents.map(createSearcheeFromTorrentFile) //also create searchee from path
 		);
 		parsedTorrents = searcheeResults.filter(ok);
+	} else if (dataDirs.length > 0) {
+		var fullPaths: string[] = [];
+		dataDirs.forEach(dataDir => {
+            const allPaths = getFilePathsFromPath(dataDir, [], 0, 4);
```

Comment:
i'm worried about the 4 here. I feel like nearly all of the matches are going to be in the 0 depth or the 1 depth - most releases aren't really nested, right?

```
datadir/
  cheers.s01e01/
    cheers.s01e01.mkv
  mash.s01/
    mash.s01e01.mkv
    mash.s01e02.mkv
    mash.s01e03.mkv
    Sample/
      mash.s01e01.sample.mkv
```

for this tree, I personally would only want searchees for the first two. I could easily see an argument for the first 6, but definitely not the last 2. 
1. `mash.s01`
2. `cheers.s01e01.mkv`
3. `mash.s01e01.mkv`
4. `mash.s01e02.mkv`
5. `mash.s01e03.mkv`
6. `cheers.s01e01`
7. `Sample`
8. `mash.s01e01.sample.mkv`

I also would still like to apply some favoritism to single-file torrents if we can - I've gone on record before as saying that I want to use my power, however small, to encourage single-file torrents and discourage multi-file torrents with extra crap in them (nfos, samples, etc). This links back up to my earlier question about `MATCH_EXCEPT_PARENT_DIR` - maybe we can support the first 2 by default, and the first 5 on risky mode if the last one (6) doesn't really have a good hit rate in practice. If it does hit regularly, then maybe not (edit: i guess I sort of forgot about packs, although I vaguely remember hearing that maybe packs aren't a thing in usenet?. 

You'll notice, also, that my priority list doesn't really relate linearly to `maxDepth` - the first one is a 0 depth, and the second one is a 1 depth. I think we can probably apply heuristics to be smart about this, but that can probably come later.

Reply:
4 is probably a little deep yes, I think my deepest match was 3 (individual episodes in season pack). 

This also calls into discussion what safe/risky should actually be. I think of them to be used for matching, not searching. For searching, the only thing to be concerned about is API hits, and IMO anyone concerned about api hits probably shouldn't be using a tool that can generate hundreds of queries and thousands of grabs on a mid-sized library.  I think just setting the max depth to 3 is a reasonable enough heuristic to get a high match rate.

In a way, single file torrents are still favored since they're more likely to be a perfect match. TBH I'm not sure if I've enabled renaming of single file torrents, but I need to do that if I haven't. I totally agree on the philosophy of single file torrents. To be fair, this still doesn't match torrents with .nfos and samples and whatnot, only single file in a directory. Of course, I think this practice should be ditched as well, but according to my results it's still a major use-case. Granted, I'm still mainly cross-seeding to places like ipt and tl, and not upper tier trackers since I'm not in them. I'm not sure how common these practices are in a tracker with better organization like BHD.

Reply:
> I think my deepest match was 3 (individual episodes in season pack).

ooc, what makes that 3 levels deep?

> For searching, the only thing to be concerned about is API hits, and IMO anyone concerned about api hits probably shouldn't be using a tool that can generate hundreds of queries and thousands of grabs on a mid-sized library

With all due respect, imo this is a lazy answer. As the developers of this API-happy tool, we especially have a duty to minimize our API footprint as much as we can, because the effects of our decisions here multiply by the size of the userbase. If that comes at the cost of a small number of matches, then it's worth it IMO.

Reply:
>ooc, what makes that 3 levels deep?
top level -> season -> per episode folder -> episode.mkv

While an uncommon structure, it unfortunately exists. 

>With all due respect, imo this is a lazy answer. As the developers of this API-happy tool, we especially have a duty to minimize our API footprint as much as we can, because the effects of our decisions here multiply by the size of the userbase. If that comes at the cost of a small number of matches, then it's worth it IMO.

Good points, I agree. 

I think it comes down to there being two safe/risky concepts: safe/risky in matching, which is what the current flag dictates, and safe/risky in api hits. The best way to limit risky api hits is by searching less of the files you come across and not going as deep in directories. Like I said, I believe my deepest matches were 3 levels deep. IMO these probably aren't worth it. I do think it makes sense to go 2 deep (single file inside folder inside parent dir) because that's how radarr libraries are laid out, which I think is a major use case. 

That said, I don't think we can simultaneously have favoritism for single file torrents while also having good results on a torrent download folder. Running on only 1 layer deep is essentially what it does now but on data instead of torrents. If the download was a folder, then it's searched as a folder. To prefer matching single-file torrents, we have to drill down to the file itself, which is frequently on layer 2.

Right now, it will search a folder that contains only 1 torrent file. I think this can easily be gotten rid of to reduce API hits by half with no losses (when running on a radarr library). We can also kick up the minimum size a bit to reduce further (2gb perhaps?), and limit the depth to 2 or 3 (I need to experiment on what this number means relative to the search dir) to reduce further. I would say this is sufficient to cover the bases without the need for a safe/risky heuristic setting for searching. With those changes, running on a data dir will probably not produce that many more API hits than if that equivalent dir were all torrents in a client.

Anyway, with your concerns about MATCH_EXCEPT_PARENT_DIR, like I said on my trackers it's extremely common,

Reply:
Very well reasoned, this all sounds good. Sorry I haven't had a lot of time to give this PR attention lately; I'm in a bit of a cross-seed hibernation cycle (I do most of my development in the winter). I do still plan on working on daemon mode support for this, but it may be a little while.